src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -174,6 +183,74 @@ public void Serialize(object value, ObjectWriter writer, CancellationToken cance
         }
     }
 
+    private static void Write(ObjectWriter writer, ImmutableDictionary<string, StructuredAnalyzerConfigOptions> optionsByLanguage)
+    {
+        // Only serialize options for C#/VB since other languages are not OOP.
+
+        var csOptions = ImmutableDictionary.GetValueOrDefault(optionsByLanguage, LanguageNames.CSharp);
+        var vbOptions = ImmutableDictionary.GetValueOrDefault(optionsByLanguage, LanguageNames.VisualBasic);
+
+        writer.WriteCompressedUInt((uint)((csOptions != null ? 1 : 0) + (vbOptions != null ? 1 : 0)));
+
+        Serialize(LanguageNames.CSharp, csOptions);
+        Serialize(LanguageNames.VisualBasic, vbOptions);
+
+        void Serialize(string language, StructuredAnalyzerConfigOptions? options)
+        {
+            if (options != null)
+            {
+                writer.WriteString(language);
+
+                // order for deterministic checksums
+                foreach (var key in options.Keys.Order())
+                {
+                    if (options.TryGetValue(key, out var value))
+                    {
+                        writer.WriteString(key);
+                        writer.WriteString(value);
+                    }
+                }
+
+                // terminator:
+                writer.WriteString(null);
+            }
+        }
+    }
+
+    private static ImmutableDictionary<string, StructuredAnalyzerConfigOptions> ReadFallbackAnalyzerOptions(ObjectReader reader)
+    {
+        var count = reader.ReadCompressedUInt();
+        if (count == 0)
+        {
+            return ImmutableDictionary<string, StructuredAnalyzerConfigOptions>.Empty;
+        }
+
+        var optionsByLanguage = ImmutableDictionary.CreateBuilder<string, StructuredAnalyzerConfigOptions>();
+        var options = ImmutableDictionary.CreateBuilder<string, string>();
+
+        for (var i = 0; i < count; i++)
+        {
+            var language = reader.ReadRequiredString();
+
+            while (true)
+            {
+                var key = reader.ReadString();
+                if (key == null)
+                {
+                    break;
+                }
+
+                var value = reader.ReadRequiredString();
```

Comment:
nit: this worries me as well.  i don't have confidence we might not have a null-value here (unless the optiosn truly do not allow that).

Reply:
Editorconfig option values are non-null strings. See https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerConfigOptions.cs#L22

Reply:
k.  i can live with this.